SCClassLibrary/Common/Collections/Dictionary.sc

```diff
@@ -493,6 +512,21 @@ IdentityDictionary : Dictionary {
 		parent = newParent;
 	}
 
+	== { arg that;
+		^this.superPerform('==', that)
+		and: { parent == that.parent }
+		and: { proto == that.proto }
+		and: { know == that.know }
+	}
+
+	hash {
+		var hash = super.hash;
+		hash = hash bitXor: know.hash;
+		hash = hash bitXor: parent.hash << 1;
+		hash = hash bitXor: proto.hash << 1;
+		^hash
```

Comment:
and then here:
```supercollider
	var hash = super.hash;
	hash = hash bitXor: know.hash;
	hash = hash bitXor: parent.hash << 1;
	hash = hash bitXor: proto.hash << 1;
	^hash
```
is probably better written as
```supercollider
	var hash = hash bitXor: know.hash;
	hash = (hash << 1) bitXor: parent.hash;
	hash = (hash << 1) bitXor: proto.hash;
	hash = (hash << 1) bitXor: super.hash;
	^hash
```
since parent, proto, and know are most likely to vary in their lower bits in typical usage. because of the particular implementation of `super.hash`, shifting it left will cause further collisions for similar arrays (see my main review comment)